File: .copilot/skills/workflow/examples/minimal-review.cwf.py
items = args or ["item one", "item two"]


def review(item):
    return wf.agent(
        f"Review this item and report concrete findings: {item}",
        agent="worker",
        phase="review",
        label=str(item)[:24],
    )


def verify(result, item):
    return {
        "item": item,
        "finding": result,
        "verdict": wf.verify(
            result,
            rubric="specific, supported, and actionable",
            phase="verify",
            label=str(item)[:24],
        ),
    }


rows = [row for row in wf.pipeline(items, review, verify) if row is not None]
kept = [row["finding"] for row in rows if row["verdict"].passed]

report = wf.synthesize(
    kept,
    prompt="Deduplicate and summarize these verified findings.",
    label="report",
)
print(report.content)

File: .copilot/skills/workflow/examples/minimal-review.cwf.mjs
// minimal-review.cwf.mjs — the smallest useful workflow: review, verify, synthesize.
export const meta = { name: "minimal-review", description: "Review items, verify findings, and synthesize a report." };

const items = args || ["item one", "item two"];

const review = (item) => agent(`Review this item and report concrete findings: ${item}`, { agentType: "worker", phase: "review", label: String(item).slice(0, 24) });

const verifyRow = async (result, item) => ({
	item,
	finding: result,
	verdict: await verify(result, "specific, supported, and actionable", { phase: "verify", label: String(item).slice(0, 24) }),
});

const rows = (await pipeline(items, review, verifyRow)).filter((row) => row !== null);
const kept = rows.filter((row) => row.verdict.passed).map((row) => row.finding);

const report = await synthesize(kept, { prompt: "Deduplicate and summarize these verified findings.", label: "report" });
return report.content;
